src/main/java/com/github/wovnio/wovnjava/SubdomainUrlLanguagePatternHandler.java

```diff
@@ -1,55 +1,76 @@
 package com.github.wovnio.wovnjava;
 
-import java.util.ArrayList;
 import java.util.regex.Pattern;
 
 class SubdomainUrlLanguagePatternHandler extends UrlLanguagePatternHandler {
     private Lang defaultLang;
-    private ArrayList<Lang> supportedLangs;
+    private LanguageAliases languageAliases;
     private Pattern getLangPattern;
 
-    SubdomainUrlLanguagePatternHandler(Lang defaultLang, ArrayList<Lang> supportedLangs) {
+    SubdomainUrlLanguagePatternHandler(Lang defaultLang, LanguageAliases languageAliases) {
         this.defaultLang = defaultLang;
-        this.supportedLangs = supportedLangs;
+        this.languageAliases = languageAliases;
         this.getLangPattern = this.buildGetLangPattern();
     }
 
     Lang getLang(String url) {
-        Lang lang = this.getLangMatch(url, this.getLangPattern);
-        return (lang != null && this.supportedLangs.contains(lang)) ? lang : null;
+        String languageIdentifier = this.resolvePatternMatch(url, this.getLangPattern);
+        Lang lang = this.languageAliases.getLanguageFromAlias(languageIdentifier);
+        if (lang != null) {
+            return lang;
+        } else if (this.languageAliases.hasAliasForDefaultLang) {
+            return null;
```

Comment:
It occurs to me that these special rules for default lang alias really only apply for path, but for consistency we have to do it I suppose.

Reply:
I ignore `hasAliasForDefaultLang` for query url patterns because I did not feel it makes sense.

For subdomain, I feel like it is possible to let it work the same way as path (as is done here). With default language English using an alias `"us"`, it will mean that
* `http://us.site.com/*` is processed as request in default language
* `http://site.com/*` is identified as outside the scope of what wovnjava will handle, and is not processed
* `http://ja.site.com/*` is processed as request for content translated into Japanese

If we feel like it would be best to make this behavior different (like I did for query pattern), that's also ok.

Reply:
I think that's OK. How about when there are no aliases at all, and they access "zz.site.com". Should it be going through the else block and returning default lang?

Reply:
Yes, that sounds correct.

Without any aliases:
```
default language     -> "zz.site.com"
translated into thai -> "th.zz.site.com"
```